### PR TITLE
Fix code folding icon color not matching defined color (3.x)

### LINKED ISSUE #58379: Script editor code folding color is influenced by main theme base color
### Godot version

3.4.2 stable

### System information

Windows 10

### Issue description

Code folding color is influenced by main theme color. It's hard to see difference when change the code folding color. But when you change the main theme color, the arrow color changes.
Use a  main dark color
image
image
Use a main light color
image
image
Change Code Folding Color doesn't do anything
image
image

### Steps to reproduce

Use a light theme and a dark theme code editor theme. Change code folding color settings in editor settings. 

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by PhoenixFlame101**
Fix #58379 for `3.x`

Cherrypick of #58457 from `master`

Caused by reusing icons from the main editor in the code editor. These
icons were converted based on the main editor theme and not the code
editor theme.

 - Create new icons for use specifically in the code editor
 - Add these icons to the exceptions when converting dark theme icons
   to light theme automatically
 - Change the default value of the code folding color to match previous
   color
 - Code folding icon is now pure white by default to correctly match
   the color defined in settings

### CODE REVIEW

File: editor/editor_themes.cpp [hunk 1]
```diff
@@ -233,6 +233,14 @@ void editor_register_and_generate_icons(Ref<Theme> p_theme, bool p_dark_theme =
...
 		exceptions.insert("StatusWarning");
 		exceptions.insert("OverbrightIndicator");
+
+		// Prevents Code Editor icons from changing
+		exceptions.insert("GuiTab");
+		exceptions.insert("GuiSpace");
+		exceptions.insert("GuiEllipsis");
+		exceptions.insert("CodeFoldedRightArrow");
+		exceptions.insert("CodeFoldDownArrow");
+		exceptions.insert("TextEditorPlay");
```
Comment (YuriSizov) [inline editor/editor_themes.cpp:243]: You never used this icon in this version.

**Comment by YuriSizov** [comment]
Seems fine other than that note and some issue that failed the static check. Needs to be rebased and fixed.

**Comment by YuriSizov** [comment]
I think you've rebased it against the master.

**Comment by PhoenixFlame101** [comment]
> I think you've rebased it against the `master`, when you should've used `3.x`.
> 
> You probably need to open a new PR at this point.

Yeah my bad, working on it 😅

**Comment by PhoenixFlame101** [comment]
I seem to have fixed the issue; rebased with 3.x and cherry-picked the commit.

Is it better to open a new PR with the same name or reopen this one? The changes I made just now aren't reflected here because the PR closed I think.

The static errors were because I forgot to run svgcleaner on two of the new icons.

**Comment by KoBeWi** [comment]
It's impossible to reopen a PR if the branch was force-pushed while it was closed. (and you can't change the branch)

**Comment by YuriSizov** [comment]
@PhoenixFlame101 Yeah, open a new one please.

**Pull request closed without merging**